linalg/hypre.cpp

```diff
@@ -1243,9 +1243,13 @@ void HypreParMatrix::Threshold(double threshold)
    row_starts = hypre_ParCSRMatrixRowStarts(A);
    col_starts = hypre_ParCSRMatrixColStarts(A);
 
+   bool old_owns_row = hypre_ParCSRMatrixOwnsRowStarts(A);
+   bool old_owns_col = hypre_ParCSRMatrixOwnsColStarts(A);
    parcsr_A_ptr = hypre_ParCSRMatrixCreate(comm,row_starts[num_procs],
                                            col_starts[num_procs],row_starts,
                                            col_starts,0,0,0);
```

Comment:
This is unrelated, but I just noticed it: it seems to me that using `row_starts[num_procs]` and `col_starts[num_procs]` here will not work in the (now default) case of "assumed partitioning". Should we switch to calling `HypreParMatrix::GetGlobalNumRows()` and `HypreParMatrix::GetGlobalNumCols()`?

@mlstowell, what do you think?

Reply:
Good point -- I think we should switch.

Reply:
Done. @v-dobrev can you approve?